React InstantSearch/injected-content/src/pages/Recommend.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+
+import algoliasearch from 'algoliasearch';
+import {
+  Configure,
+  Highlight,
+  InstantSearch,
+  SearchBox,
+} from 'react-instantsearch-dom';
+
+import recommend from '@algolia/recommend';
+import { useFrequentlyBoughtTogether } from '@algolia/recommend-react';
+
+import { InjectedInfiniteHits } from '../InjectedInfiniteHits';
+
+const searchClient = algoliasearch(
+  'XX85YRZZMV',
+  'd17ff64e913b3293cfba3d3665480217'
+);
+
+const recommendClient = recommend(
+  'XX85YRZZMV',
+  'd17ff64e913b3293cfba3d3665480217'
+);
+
+export function Recommend() {
+  return (
+    <InstantSearch
+      searchClient={searchClient}
+      indexName="test_FLAGSHIP_ECOM_recommend"
+    >
+      <Configure hitsPerPage={18} />
+      <div className="search-panel">
+        <div className="search-panel__results">
+          <SearchBox
+            className="searchbox"
+            translations={{
+              placeholder: 'Search for articles like "dress" or "jeans"',
+            }}
+          />
+          <InjectedInfiniteHits
+            slots={() => [
+              {
+                injectAt: 2,
+                slotComponent: RelatedProducts,
+              },
+            ]}
+            hitComponent={ProductHit}
+          />
+        </div>
+      </div>
+    </InstantSearch>
+  );
+}
+
+function RelatedProducts({ resultsByIndex }) {
+  const { recommendations } = useFrequentlyBoughtTogether({
+    recommendClient,
+    indexName: 'test_FLAGSHIP_ECOM_recommend',
+    objectIDs: resultsByIndex.test_FLAGSHIP_ECOM_recommend.hits
+      .slice(0, 2)
```

Comment:
why not related to more than the first two items? this seems on purpose, but I don't really understand why

Reply:
I believe it would break the grid otherwise.

Reply:
Sorry, I wasn't asking why there are two results, that makes sense, I was asking why it's related to the first two hits only

Reply:
Ah! Yes that's a good question.

Reply:
Answered to @francoischalifour below. Now the recommendations are computed for the first item and are visually linked to it.